tests/test_diffusers.py

```diff
@@ -2211,3 +2212,75 @@ def test_stable_video_diffusion_no_throughput_regression_bf16(self):
         self.assertEqual(len(outputs.frames[0]), 25)
         if IS_GAUDI2:
             self.assertGreaterEqual(outputs.throughput, 0.95 * 0.012)
+
+
+class GaudiDeterministicImageGenerationTester(TestCase):
+    """
+    Test deterministic generation using text_to_image_generation.py.
+    """
+
+    @slow
+    def test_deterministic_image_generation(self):
+        path_to_script = (
+            Path(os.path.dirname(__file__)).parent / "examples" / "stable-diffusion" / "text_to_image_generation.py"
+        )
+        install_requirements(path_to_script.parent / "requirements.txt")
+
+        with tempfile.TemporaryDirectory():
+            test_args = f"""
+                python3
+                {path_to_script}
+                --model_name_or_path runwayml/stable-diffusion-v1-5
+                --num_images_per_prompt 20
+                --batch_size 4
+                --image_save_dir /tmp/stable_diffusion_images
+                --use_habana
+                --use_hpu_graphs
+                --gaudi_config Habana/stable-diffusion
+                --bf16
+                --deterministic
```

Comment:
```suggestion
                --use_cpu_rng
```

Reply:
Thanks for the catch 👍 

I have rebased the code to adapt the newly introduced multi-card inferencing. Also corrected the cmd argument `--deterministic` in the test file.

Tested the rebased codes using the new distributed inferencing and single card examples. They are all able to generate results. I have also run the test case for GaudiDeterministicImageGenerationTester. The result is shown below:

![image](https://github.com/user-attachments/assets/ef0129c0-3d3c-4a28-9d79-3c008f2060c7)